packages/client-common/src/utils/url.ts

```diff
@@ -78,5 +80,13 @@ export function toSearchParams({
     params.set('session_id', session_id)
   }
 
+  if (role && typeof role === 'string') {
+    params.set('role', role)
+  } else if (role && Array.isArray(role)) {
+    for (const r of role) {
+      params.append('role', r)
+    }
+  }
+
```

Comment:
Nit: maybe just a single undefined check will suffice?
```suggestion
  if (role) {
    if (typeof role === 'string') {
      params.set('role', role)
    } else if (Array.isArray(role)) {
      for (const r of role) {
        params.append('role', r)
      }
    }
  }

```